code/modules/power/apc.dm

```diff
@@ -770,22 +767,17 @@
 	else
 		return APC_MALF_NOT_HACKED
 
-/obj/machinery/power/apc/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
-	if(!user)
-		return
-
-	// update the ui if it exists, returns null if no ui is passed/found
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+/obj/machinery/power/apc/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		// the ui does not exist, so we'll create a new one
-		ui = new(user, src, ui_key, "apc.tmpl", "[area.name] - APC", 510, issilicon(user) ? 535 : 460)
+		ui = new(user, src, ui_key, "APC", name, 510, issilicon(user) ? 535 : 460, master_ui, state)
```

Comment:
Why does the UI need so much extra height if they are a silicon? I don't think there is that much extra information to display if they're a silicon. The only thing that comes to mind is the malf AI overload APC button.